workflow-ui/core-common/src/main/java/com/squareup/workflow1/ui/navigation/NavigationMonitor.kt

```diff
@@ -0,0 +1,51 @@
+package com.squareup.workflow1.ui.navigation
+
+import com.squareup.workflow1.ui.Compatible
+import com.squareup.workflow1.ui.unwrap
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.onEach
+
+/**
+ * Reports navigation across a series of calls to [update], probably made
+ * for each rendering posted by
+ * [renderWorkflowIn][com.squareup.workflow1.renderWorkflowIn].
+ *
+ * Takes advantage of [unwrap()] and [Compatible.keyFor] to provide navigation
+ * logging by reporting the top (read: last-most, inner-most) sub-rendering,
+ * which conventionally is the one that is visible and accessible to the user.
+ *
+ * Reports each time the `Compatible.keyFor` the top is unequal to the previous one,
+ * which conventionally indicates that a new view object will replace the previous one.
+ */
+public class NavigationMonitor(
+  skipFirstScreen: Boolean = false,
+  private val onNavigate: (Any) -> Unit = { println(it) }
+) {
+  private var lastKey: String? = if (skipFirstScreen) null else ""
```

Comment:
@pyricau @steve-the-edwards We have only ever used our flavor of this on the main thread with the Main.Immediate dispatcher, so this simple implementation has been just fine. Should we stick with what we know? Should this field be marked `@Volatile`?

Reply:
My gut is to leave it this way. It's such a trivial class that anyone doing something more interesting and finding flaws could so easily fork it.

Should I put something in the kdoc warning of its limitations?

Reply:
Enforce that it's single threaded.

Reply:
I think that's the sanity check I was hoping for, "`@Volatile` is just fine."